File: documentapi/src/vespa/documentapi/loadtypes/loadtype.cpp
// Copyright 2016 Yahoo Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "loadtype.h"
#include <vespa/vespalib/stllike/asciistream.h>

namespace documentapi {

const LoadType LoadType::DEFAULT(0, "default", Priority::PRI_NORMAL_3);

void
LoadType::print(vespalib::asciistream & os) const
{
    os << "LoadType(" << getId() << ": " << getName() << ")";
}

} // documentapi

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/loadtypes/LoadType.java
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.documentapi.messagebus.loadtypes;

import com.yahoo.documentapi.messagebus.protocol.DocumentProtocol;

/**
 * @author thomasg
 */
public class LoadType {
    private int id;
    private String name;
    private DocumentProtocol.Priority priority;

    public static LoadType DEFAULT = new LoadType(0, "default", DocumentProtocol.Priority.NORMAL_3);

    public LoadType(int id, String name, DocumentProtocol.Priority priority) {
        this.id = id;
        this.name = name;
        this.priority = priority;
    }

    @Override
    public boolean equals(Object other) {
        if (!(other instanceof LoadType)) {
            return false;
        }

        LoadType o = (LoadType)other;

        return name.equals(o.getName()) && id == o.getId() && priority == o.getPriority();
    }

    @Override
    public int hashCode() {
        return java.util.Objects.hash(name, id, priority);
    }

    public String getName() { return name; }

    public String toString() { return name + " (id " + id + ")"; }

    public DocumentProtocol.Priority getPriority() { return priority; }

    public int getId() { return id; }
}
